OpenRA.Game/Traits/Target.cs

```diff
@@ -133,6 +133,24 @@ public WPos CenterPosition
 			}
 		}
 
+		public CPos Location
+		{
+			get
+			{
+				switch (Type)
+				{
+					case TargetType.Actor:
+						return actor.Location;
+					case TargetType.FrozenActor:
+						return frozen.Location;
+					default:
+					case TargetType.Invalid:
+					case TargetType.Terrain:
+						throw new InvalidOperationException("Attempting to query the location of a Target which is neither Actor nor FrozenActor");
```

Comment:
`AttackRequiresEnteringCell` is a hack (one of many) for the D2K sand worm.  We already know that the sandworm code needs a complete rewrite, so it would be better to keep the hacks there than to add (hopefully temporary) workarounds into the engine code.